pennylane/ops/qubit/special_unitary.py

```diff
@@ -0,0 +1,329 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the operation SpecialUnitary and
+its utility functions.
+"""
+# pylint: disable=arguments-differ
+from functools import lru_cache, reduce
+from itertools import product
+
+import numpy as np
+import pennylane as qml
+from pennylane.operation import AnyWires, Operation
+
+_pauli_matrices = np.array(
+    [[[1, 0], [0, 1]], [[0, 1], [1, 0]], [[0, -1j], [1j, 0]], [[1, 0], [0, -1]]]
+)
+"""Single-qubit Paulis:    I                 X                   Y                  Z"""
+
+_pauli_letters = "IXYZ"
+"""Single-qubit Pauli letters that make up Pauli words."""
+
+
+@lru_cache
+def pauli_basis_matrices(num_wires):
```

Comment:
We discussed this also here: https://github.com/PennyLaneAI/pennylane/pull/3650/#discussion_r1089522120

The upshot is: This is a bit specialized, because it computes a rather large,  dense tensor. In addition, the pauli module seems to require a bit of reorganizing, and it seems easier to move things there once that is done. There are quite a few formats that are being converted there and a dense tensor representation would be yet another one added. :grimacing:

Reply:
Is it fine to resolve, @rmoyard ?

Reply:
Yes thanks :+1: